packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.tsx

```diff
@@ -85,7 +85,7 @@ export const useDesktopPicker = <
     externalSlotProps: innerSlotProps?.openPickerButton,
     additionalProps: {
       disabled: disabled || readOnly,
-      onClick: actions.onOpen,
+      onClick: open ? actions.onClose : actions.onOpen,
```

Comment:
Not sure if there was a reason why the button only opens the picker, but I feel like a toggle behavior makes sense 🤔

Reply:
Make total sense to me.